nox/sessions.py

```diff
@@ -325,16 +327,18 @@ def execute(self):
 class Result:
     """An object representing the result of a session."""
 
-    def __init__(self, session, status):
+    def __init__(self, session, status, reason=None):
         """Initialize the Result object.
 
         Args:
             session (~nox.sessions.SessionRunner):
                 The session runner which ran.
             status (~nox.sessions.Status): The final result status.
+            reason (str): Additional info.
```

Comment:
Pedantry alert: `Optional[str]`

Reply:
Parameters that have a default value of `None` are implicitly `Optional`.